src/Confix.Tool/src/Confix.Library/Entities/Component/Providers/DotNet/DotnetPackageComponentProvider.cs

```diff
@@ -94,33 +94,83 @@ public async Task ExecuteAsync(IComponentProviderContext context)
     {
         var componentConfig = resources
             .FirstOrDefault(x => x.ResourceName.EndsWith(FileNames.ConfixComponent));
-        var jsonSchema = resources
-            .FirstOrDefault(x => x.ResourceName.EndsWith(FileNames.Schema));
 
-        if (componentConfig == null || jsonSchema == null)
+        if (componentConfig == null)
         {
             throw new ExitException(
-                $"[red]Could not find component definition or schema for component[/]: {component}");
+                $"[red]Could not find component definition for component[/]: {component}");
         }
 
         App.Log.ParsingComponent(componentConfig.Assembly, componentConfig.ResourceName);
         var componentConfiguration = await LoadComponentConfigurationFromAssembly(componentConfig);
 
-        await using var schemaStream = jsonSchema.GetStream();
-        var schema = await JsonSchema.FromStream(schemaStream);
-
         if (componentConfiguration.Name is null)
         {
             throw new ExitException($"Component definition is missing name: {component}");
         }
 
+        var jsonSchema = await GetJsonSchema(resources, componentConfiguration.Name);
+
         return new Component(
             Type,
             componentConfiguration.Name,
             version,
             true,
-            new[] { componentConfiguration.Name },
-            schema);
+            new[]
+            {
+                componentConfiguration.Name
+            },
+            jsonSchema);
+    }
+
+    private static async Task<JsonSchema> GetJsonSchema(
+        IReadOnlyList<DiscoveredResource> resources,
+        string component)
+    {
+        var jsonSchema = await TryGetEmbeddedJsonSchema(resources) ??
+                         await TryBuildJsonSchemaFromGraphQL(resources);
+
+        if (jsonSchema == null)
+        {
+            throw new ExitException(
+                $"[red]Could not find/build schema for component[/]: {component}");
+        }
+
+        return jsonSchema;
+    }
+
+    private static async Task<JsonSchema?> TryBuildJsonSchemaFromGraphQL(
+        IReadOnlyList<DiscoveredResource> resources)
+    {
+        var resource = resources
+            .FirstOrDefault(x => x.ResourceName.EndsWith(FileNames.SchemaGraphQL));
+
+        if (resource != null)
+        {
+            await using var graphQLSchema = resource.GetStream();
+
+            var graphqlSchema =
+                await SchemaHelpers.LoadSchemaAsync(graphQLSchema);
+
+            return graphqlSchema.ToJsonSchema().Build();
+        }
+
+        return null;
+    }
```

Comment:
I think we mix providers here. Can we not just invoke the build of the component? like we do in other places. Then it's independent from any Provider specifics

Reply:
Indeed, abandoned after discussion. The json schema of the component should be always there.